### PR TITLE
Courier Fix

### PR DESCRIPTION
**Pull request by darklordabc**
This adds a 1 second timer to courier spawning. It checks if there is an item blocking first slot in inventory, if so, sells the item (players get full refund), adds the courier, casts it immediately, adds the flying courier and casts that immediately.

### CODE REVIEW

File: game/scripts/vscripts/components/courier/courier.lua [hunk 1]
```diff
@@ -15,8 +15,25 @@ function Courier:Init ()
...
+    -- Create couriers and then cast them straight away
+    local courier = hero:AddItemByName('item_courier')
+    if courier then
+        hero:CastAbilityImmediately(courier, hero:GetPlayerID())
+    end
+    local flying = hero:AddItemByName('item_flying_courier')
+    if flying then
+        hero:CastAbilityImmediately(flying, hero:GetPlayerID())
+    end
+
```
Comment (spar36) [inline game/scripts/vscripts/components/courier/courier.lua]: Could we then do `hero:BuyItem(slot1Item)` ? I get that this works, but kinda bad user design to just have items disappear from a players inventory.

Comment (darklordabc) [inline game/scripts/vscripts/components/courier/courier.lua]: You also have to account for charges on an item.

Comment (Haganeko) [inline game/scripts/vscripts/components/courier/courier.lua]: Could also have a dummy hero spawn the courier instead of the first player...

**Comment by chrisinajar** [comment]
Is it possible to do with a dummy unit instead of a real hero? That'd be much simpler on the items...

**Comment by Honeth** [comment]
should be possible i believe it might even be possible if we give the fountain an inventory to use the fountain instead of anything else

**Comment by warpdragon** [comment]
Hell yeah!

**Comment by warpdragon** [comment]
Ok I'm confused. Wouldn't ```function GameMode:InitGameMode()
GameMode = self
InitModule(Courier)```
put the Courier in their inventory before they even pick their teams because of InitGameMode()?

**Comment by chrisinajar** [comment]
It'd probably throw errors since the heroes don't exist yet, could give it a try though.

**Comment by Chronophylos** [comment]
No `SpawnCourier()` gets called when a Hero spawn in-game.

**Pull request merged**